test/test_core.py

```diff
@@ -5900,6 +5900,17 @@ def test_fs_rename_on_existing(self, args):
       self.set_setting('FORCE_FILESYSTEM')
     self.do_runf('fs/test_fs_rename_on_existing.c', 'success', emcc_args=args)
 
+  @parameterized({
+    '': ([],),
+    'nodefs': (['-DNODEFS', '-lnodefs.js'],),
+    'noderawfs': (['-sNODERAWFS'],)
+  })
+  def test_fs_readdir_ino_matches_stat_ino(self, args):
+    if self.get_setting('WASMFS'):
+      self.set_setting('FORCE_FILESYSTEM')
+    self.do_runf('fs/test_fs_readdir_ino_matches_stat_ino.c', 'success', emcc_args=args)
+
+
```

Comment:
Too many newlines here.. flake8 would normally complain about this but we recently switched to ruff.  I wonder if ruff is missing this check?

Reply:
I believe the number of blank lines here is correct.  `test_fs_readdir_ino_matches_stat_ino()` is a top-level function so PEP8 says it should be surrounded by two blank liines.

https://peps.python.org/pep-0008/#blank-lines
> Surround top-level function and class definitions with two blank lines.

% `ruff rule E303`
## Why is this bad?
PEP 8 recommends using blank lines as follows:
- No more than two blank lines between top-level statements.
- No more than one blank line between non-top-level statements.

Note that ruff rule E303 is currently only available in `--preview` mode because of interactions with `isort` and typing stub files (.pyi).

I sense this returns us to the code formatting questions of #14449.  I see that [`yapf`](https://github.com/search?q=repo%3Aemscripten-core%2Femscripten%20yapf) configuration is present in the codebase but is it run on a consistant basis?  If we ran yapf or psf/black or ruff format on a consistant basis then the code formatter could autofix blank lines instead of it being a linter's responsibility.
* #14449

Reply:
It's not a top level function though it's a class method.

Reply:
> If we ran yapf or psf/black or ruff format on a consistent basis then the code formatter could autofix blank lines instead of it being a linter's responsibility.

I think we would always want the linter running CI just in case developers don't run the formatter locally.

I confirmed locally that ruff is not enforcing black line rules today, so we do have a linting regression here.

Reply:
* #23154

Reply:
Can we focus on making sure that `ruff check` reports when the wrong number of blank lines are using?

I'm not complete averse to a full code reformat, but that is much larger change that I think will require a fair amount of dicussion/time... and it seem orthogonal to the fact that `ruff check` didn't report this issue.

Reply:
* #23155